src/sparseml/exporters/transforms/base_transform.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class BaseTransform(ABC):
+    """
+    Interface that all transforms must implement.
+    A transform is an object that, when applied to a model,
+    does a single modification to it
+    and returns a modified model.
+    """
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def __call__(self, model: Any) -> Any:
+        """
+        1. Validate the input model
+        2. Apply the transform to the model
+        3. Validate the resulting model and return it
+
+        :param model: The input model to be transformed
+        :return: The transformed model
+        """
+        self._validate_input(model)
```

Comment:
Maybe we can add a private function that maps from any input (model object, path string) to the actual model object e.g.`_read_the_model`? This way we might specify all the functionalities of the `__call__` method in the base class and would rarely have a need to modify it in the children classes.

Reply:
I like this idea but let's put off refactors like it until we actually have a good set of transforms. Don't want to make assumptions like this without actually having many of the transforms built